packages/teleport-generator-shared/src/utils/project-utils.ts

```diff
@@ -248,14 +249,144 @@ export const createFile = (name: string, fileType: string, content: string): Gen
   return { name, fileType, content }
 }
 
-export const createFolder = (
-  name: string,
-  files: GeneratedFile[] = [],
-  subFolders: GeneratedFolder[] = []
+interface LocalDependenciesMeta {
+  defaultComponentsPath: string | string[]
+  defaultPagesPath: string | string[]
+}
+
+export const generateLocalDependenciesPrefix = (
+  template: TemplateDefinition,
+  meta: LocalDependenciesMeta
+): string => {
+  const initialComponentsPath =
+    template.meta && template.meta.componentsPath
+      ? template.meta.componentsPath
+      : [].concat(meta.defaultComponentsPath)
+
+  const initialPagesPath =
+    template.meta && template.meta.pagesPath
+      ? template.meta.pagesPath
+      : [].concat(meta.defaultPagesPath)
+
+  let dependencyPrefix = ''
+
+  /*
+    Remove common path elements from the beginning of the
+    components and pages full path (if any)
+  
+    For example, having:
+    - initialComponentsPath = ['src', 'components']
+    - initialPagesPath = ['src', 'pages']
+  
+    If we want to have an import statement that goes from the pages folder to the
+    components folder, we only need to go back one step, so we are removing
+    the forst element from both the paths ('src') and build the dependencyPrefix accordingly
+  */
+  const { componentsPath, pagesPath } = removeCommonStartingPointsFromPaths(
+    initialComponentsPath,
+    initialPagesPath
+  )
+
+  // We have to go back as many folders as there are defined in the pages path
+  dependencyPrefix += '../'.repeat(pagesPath.length)
+
+  dependencyPrefix += componentsPath
+    .map((component) => {
+      return `${component}/`
+    })
+    .join('')
+
+  return dependencyPrefix
+}
+
+const removeCommonStartingPointsFromPaths = (
+  componentsPath: string[],
+  pagesPath: string[]
+): { componentsPath: string[]; pagesPath: string[] } => {
+  const componentsPathLength =
+    componentsPath.length > pagesPath.length ? componentsPath.length : pagesPath.length
+
+  let i = 0
+  let isEqual = true
```

Comment:
i and isEqual are a bit too generic in this instance, you need to look at the code in depth to understand what's their meaning

Reply:
Renamed:
i -> componentIndex
isEqual -> elementsFromIndexAreEqual